specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -84,6 +84,10 @@ model AnalyzeTextResult {
   @added(ContentSafety.Versions.v2023_10_30_Preview)
   @doc("The incident match details.")
   incidentMatches?: IncidentMatch[];
+
+  @added(ContentSafety.Versions.v2023_11_30_Preview)
+  @doc("Sentences detected as harmful content")
+  reason?: string[];
```

Comment:
As I proposed yesterday, we need to consider the concepts of "reason"/"ungroundedTextReason"/"reasoning". To my understanding they are different things now:
1. In explanation, reason means the array of text that are harmful.
2. In ungroundedness detect, ungroundedTextReason explains why the text is ungrounded, but not the text itself.
3. In annotation service, reasoning is the statements that defined in the label that indicates why the text is annotated as label.

So if they share the same root "reason", it may cause confusion in the future.